UI/org.eclipse.birt.report.designer.ui/src/org/eclipse/birt/report/designer/internal/ui/views/SearchInputDialog.java

```diff
@@ -0,0 +1,599 @@
+		gd = new GridData(SWT.FILL, SWT.FILL, true, true);
+		gd.widthHint = 250;
+		gd.horizontalSpan = 2;
+		tableContainer.setLayoutData(gd);
+
+		layout = new GridLayout();
+		layout.numColumns = 1;
+		layout.marginWidth = layout.marginHeight = 0;
+		tableContainer.setLayout(layout);
+
+		final Table table = new Table(tableContainer, SWT.VIRTUAL | SWT.BORDER);
+		table.setLinesVisible(true);
+		table.setHeaderVisible(true);
+		gd = new GridData(SWT.FILL, SWT.FILL, true, true);
+		gd.heightHint = 100;
+		table.setLayoutData(gd);
+
+		TableColumn pathColumn = new TableColumn(table, SWT.NONE);
+		pathColumn.setText("Property");
+		table.getColumn(0).pack();
+
+		TableColumn propColumn = new TableColumn(table, SWT.NONE);
+		propColumn.setText("Element");
+		table.getColumn(1).pack();
+
+//		table.setItemCount(6);
+//		table.addListener(SWT.SetData, new Listener() {
+//			public void handleEvent(Event event) {
+//				TableItem item = (TableItem) event.item;
+//				int index = table.indexOf(item);
+//				item.setText(0, "Item " + index);
+//				item.setText(1, "foobar");
+//			}
+//		});
+
+		table.addListener(SWT.Selection, event -> {
+			String string = event.detail == SWT.CHECK ? "Checked" : "Selected";
+			System.out.println(event.item + " " + string);
+			this.itemSelected(event.item.getData());
+		});
+		this.table = table;
+
+		setErrorMessage(errorMessage);
+
+		applyDialogFont(composite);
+
+		UIUtil.bindHelp(parent, helpContextID);
+		loadPropCombo();
+
+		return composite;
+	}
+
+	/**
+	 *
+	 */
+	private void loadPropCombo() {
+		String text = propCombo.getText();
+		propCombo.removeAll();
+		propCombo.add("(any)");
+		boolean found = text.equals("(any)");
+		if (services != null) {
+			boolean recursive = this.recursiveButton.getSelection();
+			List<String> list = services.getPropertyNames(recursive);
+			for (String name : list) {
+				propCombo.add(name);
+				if (!found && text.trim().length() > 0 && name.equals(text)) {
+					found = true;
+				}
+			}
+		}
+		if (!found) {
+			text = "(any)";
+		}
+		propCombo.setText(text);
+	}
+
+	/**
+	 * @param errorMessage
+	 */
+	public void setErrorMessage(String errorMessage) {
+		this.errorMessage = errorMessage;
+		if (errorMessageText != null && !errorMessageText.isDisposed()) {
+			errorMessageText.setText(errorMessage == null ? " \n " : errorMessage); //$NON-NLS-1$
+
+			boolean hasError = errorMessage != null && (StringConverter.removeWhiteSpaces(errorMessage)).length() > 0;
+			errorMessageText.setEnabled(hasError);
+			errorMessageText.setVisible(hasError);
+			errorMessageText.getParent().update();
+		}
+	}
+
+	@Override
+	protected Control createContents(Composite parent) {
+		Control composite = super.createContents(parent);
+		if (valueText.getText().trim().length() == 0) {
+			getButton(SEARCH_ID).setEnabled(false);
+		}
+		getButton(BACK_ID).setEnabled(false);
+		return composite;
+	}
+
+	/**
+	 * @since 3.3
+	 *
+	 */
```

Comment:
The whole class is new so no need to put this on nested content.

Reply:
If you are referring to overridden createContents, I copied this from RenameInputDialog and I'm not sure how else to do it.

Reply:
If there is an `@since` it must be 4.20.   If the containing context/class already has an `@since` then it's not needed because the container is new and then you don't need to add `@since` to each child.